### DISCUSSION TITLE
Recipes has no side effects

### DISCUSSION THREAD

**Discussion by unilogica** [chakra-ui/chakra-ui#9398] [General] [answered]
I tested something very basic, like changing the bg to red in:

https://github.com/chakra-ui/chakra-ui/blob/main/packages/react/src/theme/recipes/input.ts

```
base: {
  bg: 'red'
}
```

```
const config = defineConfig({
  theme: {
    recipes: {
      input: inputRecipe,
    },
...
```

And this has no side effect anywhere. Here shouldn't every Input element of Chakra-ui inherit the bg property from the base?

**Comment by unilogica** [comment, marked as answer]
After some days of work and pain reading and testing, this solution worked for me:

```
export const system = createSystem(defaultConfig, {
  disableLayers: true,
})
```
Docs ref: https://www.chakra-ui.com/docs/styling/cascade-layers

**Comment by maulik13** [comment]
Variants will override your base color, if specified. Did you check that?

**Reply by veloware** [reply]
@unilogica ^^^ this would be more correct rather than disabling the cascading of styles, you are probably going to run into a lot of styling weirdness in time by disabling this.

There is a default variant of 'outline' -

```
outline: {
  bg: "transparent",
  borderWidth: "1px",
  borderColor: "border",
  focusVisibleRing: "inside",
},

...

defaultVariants: {
    size: "md",
    variant: "outline",
  },
```

therefore the `bg` would be set as transparent as it has a higher precedence than `base`, therefore setting a `bg` property in `base` will be overridden.

Just to point something else out though on the same example, if for example you wanted to change the border color, you can see above the `outline` variant's `borderColor` is 'border' - this is a semantic token, see here - https://github.com/chakra-ui/chakra-ui/blob/main/packages/react/src/theme/semantic-tokens/colors.ts , note the value for `border.DEFAULT`

So in this case to override the border color, you could provide a new color for the `border.DEFAULT` token in your theme, which will adjust all of the components that use the 'border' token, rather than trying to override it in each of the component themes

**Reply by unilogica** [reply]
I tested everything, and I also paid attention to the default variant (outline). The problem is that the recipes don't change anything without disabling this flag - I spent days on this issue, the only property that it is actually respecting is the className, I open the inspector and see that the elements with recipe receives what I specified only in this property (className), everything else is ignored.

"@chakra-ui/react": "^3.2.3",
"@emotion/react": "^11.14.0",

**Reply by maulik13** [reply]
@unilogica it sounds like you are not customizing the theme correctly. 

Try this:
```typescript
// -- imports --
export const inputRecipe = defineRecipe({
  base: {
    borderRadius: "24px",
  },
});

export const system = createSystem(defaultConfig, {
  theme: {
    recipes: {
      input: inputRecipe
    }
  }
});
```

And use the system in your `<ChakraProvider>`

If you want, you can take a look at how everything works together in a complete theme by running 
`npx @chakra-ui/cli eject --outdir ./theme`

Ref: https://www.chakra-ui.com/docs/theming/customization/overview

**Reply by unilogica** [reply]
Yes, I'm basing my theme on
`npx @chakra-ui/cli eject --outdir ./theme`

The test is exactly that, absolutely nothing changed in `Input` element, the strange thing is that other third-party components based on it worked, like Select. I spent days and days looking for a solution, the only way was just with `disableLayers: true`, but I appreciate the interaction.

And I can't just start another project from scratch, because there are many components from v2 migrated to v3, but I read and reread the documentation, exhausted the topics and GPT here, all tricks and the cards up my sleeve.

**Reply by maulik13** [reply]
@unilogica I understand the frustration if you have tried everything possible. Since the project is a migration it could be something to do with that. I assume you have tried doing a simple minimal test on a fresh install to compare and see the difference in styles between generated between your migrated project and a brand new project.
